Mathlib/Algebra/MonoidAlgebra/Degree.lean

```diff
@@ -222,35 +223,306 @@ variable [Semiring R]
 
 section SupDegree
 
-variable [AddZeroClass A] [SemilatticeSup B] [AddZeroClass B] [OrderBot B]
-  (D : A → B)
+variable [AddZeroClass A] [Add B] (D : A → B)
+
+section SemilatticeSup
+
+variable [SemilatticeSup B] [OrderBot B]
 
 /-- Let `R` be a semiring, let `A, B` be two `AddZeroClass`es, let `B` be an `OrderBot`,
 and let `D : A → B` be a "degree" function.
 For an element `f : R[A]`, the `AddMonoidAlgebra R A`, the element `supDegree f : B` is the
 supremum of all the elements in the support of `f`, or `⊥` if `f` is zero.
 Often, the Type `B` is `WithBot A`,
 If, further, `A` has a linear order, then this notion coincides with the usual one,
-using the maximum of the exponents. -/
-@[reducible]
-def supDegree (f : AddMonoidAlgebra R A) : B :=
-  f.support.sup D
+using the maximum of the exponents.
+
+If `A := σ →₀ ℕ` then `R[A] = MvPolynomial σ R`, and if we equip `σ` with a linear order then
+the induced linear order on `Lex A` equips `MvPolynomial` ring with a
+[monomial order](https://en.wikipedia.org/wiki/Monomial_order) (i.e. a linear order on `A`, the
+type of (monic) monomials in `R[A]`, that respects addition). We make use of this monomial order
```

Comment:
@adomani I think I'm still talking about the order on `A` here, so it would be appropriate to talk about addition on `A`, rather than multiplication on the type of monomials; I added a comma to clarify this. Maybe I should say "the type representing monomials in R[A]" or something like that.